sentry-android-core/src/main/java/io/sentry/android/core/ActivityLifecycleIntegration.java

```diff
@@ -499,15 +496,34 @@ private void cancelTtfdAutoClose() {
   }
 
   private void onFirstFrameDrawn(final @Nullable ISpan ttidSpan) {
-    if (options != null && ttfdSpan != null && ttfdSpan.isFinished()) {
+    if (options != null && ttidSpan != null) {
       final SentryDate endDate = options.getDateProvider().now();
-      ttfdSpan.updateEndDate(endDate);
+      final long durationNanos = endDate.diff(ttidSpan.getStartDate());
+      ttidSpan.setMeasurement("time-to-initial-display", durationNanos, NANOSECOND);
```

Comment:
`l`: maybe let's extract the measurement keys into constants (could be even a separate class for measurements that we'll add in the future, such as INP)?

Reply:
I agree
Let's add also the other measurements to this new class, too, like `frames_total`, `frames_slow` or `frames_frozen`

Reply:
I moved the keys as constants in `MeasurementValue`